src/components/button/button.test.tsx

```diff
@@ -0,0 +1,330 @@
+
+    expect(button).toBeDisabled();
+  });
+
+  /* Styling test for coverage */
+  it("sets the correct styling attributes when the 'destructive' and 'disabled' props are 'true'", () => {
+    render(
+      <Button destructive disabled>
+        foo
+      </Button>
+    );
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveStyleRule("color: var(--colorsActionMajorYin030)");
+    expect(button).toHaveStyleRule(
+      "background-color: var(--colorsActionMajorYin030)"
+    );
+    expect(button).toHaveStyleRule(
+      "background: var(--colorsActionDisabled500)"
+    );
+  });
+
+  /* Styling test for coverage */
+  it("renders with the correct white-space when the 'noWrap' prop is 'true'", () => {
+    render(
+      <Button aria-label="button" noWrap>
+        foo
+      </Button>
+    );
+
+    const button = screen.getByRole("button");
+    expect(button).toHaveStyle("white-space: nowrap");
+  });
+
+  it("renders an anchor element when a custom value is passed to the 'href' prop", () => {
+    render(<Button href="https://www.warnerbros.com/movies/heat">bar</Button>);
+
+    const button = screen.getByRole("link");
+
+    expect(button).toBeVisible();
+  });
+
+  it("sets the 'target' attribute correctly when a custom value is passed to the 'target' prop", () => {
+    render(
+      <Button href="https://www.warnerbros.com/movies/heat" target="_blank">
+        bar
+      </Button>
+    );
+
+    const button = screen.getByRole("link");
+
+    expect(button).toHaveAttribute("target", "_blank");
+  });
+
+  it("sets the 'rel' attribute correctly when a custom value is passed to the 'rel' prop", () => {
+    render(
+      <Button
+        href="https://www.warnerbros.com/movies/heat"
+        rel="noopener noreferrer"
+      >
+        bar
+      </Button>
+    );
+
+    const button = screen.getByRole("link");
+
+    expect(button).toHaveAttribute("rel", "noopener noreferrer");
+  });
+
+  it("renders a tooltip, populated with a custom value that is passed to the 'iconTooltipMessage' prop", async () => {
+    render(<Button iconType="bin" iconTooltipMessage="foo" />);
+
+    const user = userEvent.setup();
+    const button = screen.getByRole("button");
+    await user.hover(button);
+    const tooltip = await screen.findByText("foo");
+
+    expect(tooltip).toBeInTheDocument();
+  });
+
+  it("sets the 'id' attribute correctly when a custom value is passed to the 'id' prop", () => {
+    render(<Button id="foo">bar</Button>);
+
+    const button = screen.getByRole("button");
+
+    expect(button).toHaveAttribute("id", "foo");
+  });
+
+  it("sets the text content of the 'subtext' element when the 'size' prop is 'large' and the 'subtext' prop is passed", () => {
+    render(
+      <Button size="large" subtext="some context about the foo">
+        foo
+      </Button>
+    );
+
+    const subtext = screen.getByText("some context about the foo");
+    expect(subtext).toBeInTheDocument();
```

Comment:
**suggestion/nitpick:** You seem to have used this format elsewhere. Just to keep things consistent in the file.

```suggestion
    const subtext = screen.getByText("some context about the foo");
    
    expect(subtext).toBeInTheDocument();
```